iset.mm

```diff
@@ -151091,6 +151259,223 @@ the additive structure must be abelian (see ~ ringcom ), care must be
   $}
 
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Local rings
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c LRing $.
+  $( Extend class notation with class of all local rings. $)
+  clring $a class LRing $.
+
+  ${
+    $d x y r $.
+    $( A local ring is a non-zero ring where any sum which equals one has one
+       of the two summands which has a multiplicative inverse.  Any field is a
+       local ring; the ring of integers is an example of a ring which is not a
+       local ring.  (Contributed by Jim Kingdon, 18-Feb-2025.) $)
+    df-lring $a |- LRing = { r e. Ring | ( ( 1r ` r ) =/= ( 0g ` r ) /\
+      A. x e. ( Base ` r ) A. y e. ( Base ` r )
+      ( ( x ( +g ` r ) y ) = ( 1r ` r ) ->
+      ( x e. ( Unit ` r ) \/ y e. ( Unit ` r ) ) ) ) } $.
+  $}
+
+  ${
+    $d R r x y $.
+    $( A local ring is a ring.  (Contributed by Jim Kingdon, 20-Feb-2025.) $)
+    lringring $p |- ( R e. LRing -> R e. Ring ) $=
+      ( vx vy vr clring wcel crg cur cfv c0g wne cv cplusg co wceq cui cbs wral
+      wo fveq2 wi neeq12d oveqd eqeq12d eleq2d orbi12d imbi12d anbi12d df-lring
+      wa raleqbidv elrab2 simplbi ) AEFAGFAHIZAJIZKZBLZCLZAMIZNZUNOZUQAPIZFZURV
+      BFZSZUAZCAQIZRZBVGRZUJZDLZHIZVKJIZKZUQURVKMIZNZVLOZUQVKPIZFZURVRFZSZUAZCV
+      KQIZRZBWCRZUJVJDAGEVKAOZVNUPWEVIWFVLUNVMUOVKAHTZVKAJTUBWFWDVHBWCVGVKAQTZW
+      FWBVFCWCVGWHWFVQVAWAVEWFVPUTVLUNWFVOUSUQURVKAMTUCWGUDWFVSVCVTVDWFVRVBUQVK
+      APTZUEWFVRVBURWIUEUFUGUKUKUHBCDUIULUM $.
+  $}
+
+  ${
+    $d R r x y $.
+    $( A local ring is a non-zero ring.  (Contributed by Jim Kingdon,
+       20-Feb-2025.) $)
+    lringnz $p |- ( R e. LRing -> ( 1r ` R ) =/= ( 0g ` R ) ) $=
+      ( vx vy vr clring wcel cur cfv c0g wne cv cplusg co wceq cui wo cbs fveq2
+      wi wral wa neeq12d oveqd eqeq12d eleq2d orbi12d imbi12d raleqbidv anbi12d
+      crg df-lring elrab2 simprbi simpld ) AEFZAGHZAIHZJZBKZCKZALHZMZUPNZUSAOHZ
+      FZUTVDFZPZSZCAQHZTZBVITZUOAUJFURVKUAZDKZGHZVMIHZJZUSUTVMLHZMZVNNZUSVMOHZF
+      ZUTVTFZPZSZCVMQHZTZBWETZUAVLDAUJEVMANZVPURWGVKWHVNUPVOUQVMAGRZVMAIRUBWHWF
+      VJBWEVIVMAQRZWHWDVHCWEVIWJWHVSVCWCVGWHVRVBVNUPWHVQVAUSUTVMALRUCWIUDWHWAVE
+      WBVFWHVTVDUSVMAORZUEWHVTVDUTWKUEUFUGUHUHUIBCDUKULUMUN $.
+  $}
+
+  ${
+    $d .+ u v $.  $d ./ u v $.  $d R r u v $.  $d X u v $.  $d Y u v $.
+    lring.b $e |- ( ph -> B = ( Base ` R ) ) $.
+    lring.u $e |- ( ph -> U = ( Unit ` R ) ) $.
+    lring.p $e |- ( ph -> .+ = ( +g ` R ) ) $.
+    lring.m $e |- ( ph -> .x. = ( .r ` R ) ) $.
+    lring.d $e |- ( ph -> ./ = ( /r ` R ) ) $.
+    lring.l $e |- ( ph -> R e. LRing ) $.
+    lring.s $e |- ( ph -> ( X .+ Y ) e. U ) $.
+    lring.x $e |- ( ph -> X e. B ) $.
+    lring.y $e |- ( ph -> Y e. B ) $.
+    $( If the sum of two elements of a local ring is invertible, then one of
```

Comment:
"then **at least** one": strictly speaking, it's redundant, but it may avoid any ambiguities.

Reply:
Somehow "at least" takes my mind to cardinality rather than disjunction, so I'm not completely sure this is an improvement, but I'll accept the suggestion. The two wordings do mean the same thing here, so I don't think we'd particularly go wrong either way.

Reply:
Done.